irobotclient/interface.py

```diff
@@ -0,0 +1,117 @@
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import os
+import collections
+
+CRAM = {"data": ".cram", "index": {".crai"}}
+BAM = {"data": ".bam", "index": {".bai", ".pbi"}}
+
+class Interface:
+
+    def __init__(self, command_line_args):
+
+        self._input_file = self._remove_file_extension(command_line_args.input_file)
+        self._output_file = command_line_args.output_file
+        self._file_extensions = self._get_file_extensions(command_line_args.input_file)
+
+        if command_line_args.url is not None:
+            self._irobot_url = command_line_args.url
+        else:
+            try:
+                self._irobot_url = os.getenv("IROBOT_URL")
+                assert self._irobot_url is not None
+            except:
+                print("Cannot establish iRobot URL")
+                exit()
+
+        if command_line_args.token is not None:
+            self._arvados_token = command_line_args.token
+        else:
+            try:
+                self._arvados_token = os.getenv("ARVADOS_TOKEN")
+                assert self.arvados_token is not None
+            except:
+                print("Cannot establish Arvados authentication token")
+                exit()
+
+    def _get_file_extensions(self, input_file, no_index=False):
+        """
+
+        :param input_file:
+        :return: associated file extensions for the type of
+        """
+        split_values = input_file.split('.')
+        extension = "." + split_values[1]
+
+        print(extension) # Beth: debug
+
+        file_extensions = {}
+
+        # TODO - Convert file extensions into a list to return
+        if not no_index:
+            if extension == CRAM["data"]:
+                file_extensions = CRAM
+            elif extension == BAM["data"]:
+                file_extensions = BAM
+            else:
+                print("ERROR: Unknown input file extension")
+                exit()
+        else:
+            if extension == CRAM["data"]:
+                file_extensions = CRAM["data"]
+            elif extension == BAM["data"]:
+                file_extensions = BAM["data"]
+            else:
+                print(("ERROR: Unknown input file extension"))
+                exit()
+
+        print(file_extensions) # Beth - debug
+
+        return file_extensions
+
+    def _remove_file_extension(self, input_file):
+        """
+        Remove the file extension so requests for different files types is easier to process
+
+        :param input_file:
+        :return: file name without the extension
+        """
+
+        split_values = input_file.split('.')
+        return split_values[0]
```

Comment:
Python has neat unpacking syntax, so this can be simplified to something like:

```
stripped_filename, *_extensions = input_file.split('.')
return stripped_filename
```

That being said, this function makes the assumption that all files have the form `filename.extension` and that directories can't have dots in their name. This is not the case. Files can have no extension (e.g., `python`), multiple extensions (e.g., `backup.tar.gz`) and misleading extensions (e.g., `this_is_a_text_file.jpg`). I think you can ignore the latter case and assume the extension (if there is one) is being used correctly, but you need additional logic to deal with these other cases. For example, if your input filename was:
```
/path/to/my.foo/file.cram
```

...your function would reduce it to:
```
/path/to/my
```

...and it would fail unnecessarily.